homeassistant/components/binary_sensor/wirelesstag.py

```diff
@@ -0,0 +1,208 @@
+
+    platform.install_push_notifications(sensors)
+    add_devices(sensors, True)
+
+
+class WirelessTagBinarySensor(WirelessTagBaseSensor, BinarySensorDevice):
+    """A binary sensor implementation for WirelessTags."""
+
+    @classmethod
+    def allowed_sensors(cls, tag):
+        """Return list of allowed sensor types for specific tag type."""
+        sensors_map = {
+            # 13-bit tag - allows everything but not light and moisture
+            WIRELESSTAG_TYPE_13BIT: [
+                SENSOR_PRESENCE, SENSOR_BATTERY,
+                SENSOR_MOTION, SENSOR_DOOR,
+                SENSOR_COLD, SENSOR_HEAT,
+                SENSOR_DRY, SENSOR_WET],
+
+            # Moister/water sensor - temperature and moisture only
+            WIRELESSTAG_TYPE_WATER: [
+                SENSOR_PRESENCE, SENSOR_BATTERY,
+                SENSOR_COLD, SENSOR_HEAT,
+                SENSOR_MOISTURE],
+
+            # ALS Pro: allows everything, but not moisture
+            WIRELESSTAG_TYPE_ALSPRO: [
+                SENSOR_PRESENCE, SENSOR_BATTERY,
+                SENSOR_MOTION, SENSOR_DOOR,
+                SENSOR_COLD, SENSOR_HEAT,
+                SENSOR_DRY, SENSOR_WET,
+                SENSOR_LIGHT]
+        }
+
+        # allow everything if tag type is unknown
+        # (i just dont have full catalog of them :))
+        tag_type = tag.tag_type
+        fullset = SENSOR_TYPES.keys()
+        return sensors_map[tag_type] if tag_type in sensors_map else fullset
+
+    def __init__(self, api, tag, sensor_type):
+        """Initialize a binary sensor for a Wireless Sensor Tags."""
+        super().__init__(api, tag)
+        self._sensor_type = sensor_type
+        self.define_name('{0} {1}'.format(self._tag.name,
+                                          SENSOR_TYPES[self._sensor_type][0]))
+        self._device_class = SENSOR_TYPES[self._sensor_type][1]
+        self._tag_attr = SENSOR_TYPES[self._sensor_type][2]
+        self.binary_spec = SENSOR_TYPES[self._sensor_type][3]
+        self.tag_id_index_template = SENSOR_TYPES[self._sensor_type][4]
+        self._state = self.updated_state_value()
+        self._api.register_entity(self)
+
+    @property
+    def is_on(self):
+        """Return True if the binary sensor is on."""
+        return self._state == STATE_ON
+
+    @property
+    def device_class(self):
+        """Return the class of the binary sensor."""
+        return self._device_class
+
+    @property
+    def principal_value(self):
+        """Return value of tag.
+
+        Subclasses need override based on type of sensor.
+        """
+        return (
+            STATE_ON if getattr(self._tag, self._tag_attr, False)
+            else STATE_OFF)
+
+    def updated_state_value(self):
+        """Use raw princial value."""
+        return self.principal_value
+
+    def update_tag_info(self, event):
+        """Handle push notification sent by tag manager."""
+        pass
+
+    def on_binary_event(self, event):
```

Comment:
If we get push notifications, you should have the `should_poll` property return False.

Reply:
@balloob Well, I actually have both - local push notifications and cloud polling in place. As it stated in description of [`Classification of the Internet of Things`](https://www.home-assistant.io/blog/2016/02/12/classifying-the-internet-of-things/#classifiers) Disadvantage of `local push` is `If it does not also support polling, home automation will not be made aware of the state after booting up until it changes.` So I do support both ways supplementing each others. Actual network activity is not happen every time update is called - it is controlled by polling interval which is controlled and defined by end user at her my.wirelesstag.net account. lib respects this and keep cache of last updated values.
Additionally, there are situations present when local push is not installed - in particular if end user have not defined at least one binary sensor to have push notifications registered (I've warned on it in supplementary documentation here https://github.com/home-assistant/home-assistant.github.io/pull/5035)

Reply:
The first part could be circumvented by only polling the cloud once at startup, after that just rely on the pushes to come in.

If you control polling yourself (using `hass.helpers.async_track_interval`), you could disable polling whenever push is active and disable it otherwise. We wouldn't want to add unnecessary usage on the wireless tag API.

Reply:
I do not control polling myself relying on hass to do it for me. There are no unnecessary network traffic happens as library effectively manages time to update using polling interval defined by end user inside wireless tags hw - any calls of library in between polling interval returns cached data. Also I think keeping polling in place make solution more robust - I could imagine scenario when tags reconfigured outside of hass on other url calls which would just stop any push notifications to hass. So polling would continue to provide data... Does it make sense from your prospective?